src/core_scan/Core_scan.ml

```diff
@@ -1161,12 +1173,14 @@ let scan ?match_hook config ((valid_rules, invalid_rules), rules_parse_time) :
 (* Entry point *)
 (*****************************************************************************)
 
+let get_rules config =
+  Common.with_time (fun () -> rules_from_rule_source config)
```

Comment:
why not put the [@@trace] in rules_from_rule_source instead?
so you don't need this wrapper.